RHDEVDOCS-5248: Pruner configuration documentation (#60817)
openshift/openshift-docs: modules/op-default-pruner-configuration.adoc

[id="default-pruner-configuration_{context}"]
= Configuring the pruner

You can use the `TektonConfig` custom resource to configure periodic pruning of resources associated with pipeline runs and task runs.

The following example corresponds to the default configuragion:

.Example of the pruner configuration
[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
# ...
spec:
  pruner:
    resources:
      - taskrun
      - pipelinerun
    keep: 100
    prune-per-resource: false
    schedule: "* 8 * * *"
# ...
----

.Supported parameters for pruner configuration
|===
| Parameter | Description

|`schedule`
|The Cron schedule for running the pruner process. The default schedule runs the process at 08:00 every day. For more information about the Cron schedule syntax, see link:https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax[Cron schedule syntax] in the Kubernetes documentation.

|`resources`
|The resource types to which the pruner applies. The available resource types are `taskrun` and `pipelinerun`

|`keep`
|The number of most recent resources of every type to keep.

|`prune-per-resource`

> |If set to `false`, the number for the `keep` setting is the total number of task runs or pipeline runs. For example, if `keep` is set to `100`, then the pruner keeps 100 most recent task runs and 100 most recent pipeline runs and removes all others.

Reviewer: ```suggestion
|If set to `false`, the value of the `keep` parameter denotes the total number of task runs or pipeline runs. For example, if `keep` is set to `100`, then the pruner keeps 100 most recent task runs and pipeline runs and removes rest of the resources.
```

> If set to `true`, the number for the `keep` setting is calculated separately for pipeline runs referencing each pipeline and for task runs referencing each task. For example, if `keep` is set to `100`, then the pruner keeps 100 most recent pipeline runs for `Pipeline1`, 100 most recent pipeline runs for `Pipeline2`, 100 most recent task runs for `Task1`, and so on, and removes all others.

Reviewer: ```suggestion
If set to `true`, the value of the `keep` parameter is calculated separately for pipeline runs referencing each pipeline and for task runs referencing each task. For example, if `keep` is set to `100`, then the pruner keeps 100 most recent pipeline runs for `Pipeline1`, 100 most recent pipeline runs for `Pipeline2`, 100 most recent task runs for `Task1`, and so on, and removes rest of the resources.
```

Reviewer: @mramendi This one is not resolved.

Author: thanks fixed

|`keep-since`

> |The maximum age of resources to keep, in minutes. For example, to retain resources which were created not more than five days ago, set `keep-since` to `7200`.

Reviewer: ```suggestion
|The maximum time for resources to keep, which is set in minutes. For example, to retain resources which were created not more than five days ago, set `keep-since` to `7200`.
```
Generally, we use age for the attribution of human characteristics (instance of anthropomorphism).

|===

[NOTE]
====

> The `keep` and `keep-since` keys are mutually exclusive. Use only one of them in your configuration.

Reviewer: ```suggestion
The `keep` and `keep-since` parameters are mutually exclusive. Use only one of them in your configuration.
```
